overlord/ifacestate/handlers.go

```diff
@@ -303,3 +306,24 @@ func (m *InterfaceManager) doDisconnect(task *state.Task, _ *tomb.Tomb) error {
 	setConns(st, conns)
 	return nil
 }
+
+func (m *InterfaceManager) doDiscardNamespace(task *state.Task, _ *tomb.Tomb) error {
```

Comment:
To some extent yes but you can envision a sister task that lives right next to this one that goes into a namespace to reconfigure it based on interface changes. I think we can move it to the `snapcore` package but depending on how things unfold it might "go back" here.